13/andreitch/directors.py

```diff
@@ -0,0 +1,71 @@
+import csv
+from collections import defaultdict, namedtuple
+from functools import reduce
+
+MOVIE_DATA = '../movie_metadata.csv'
+NUM_TOP_DIRECTORS = 20
+MIN_MOVIES = 4
+MIN_YEAR = 1960
+
+Movie = namedtuple('Movie', 'title year score')
+
+
+def get_movies_by_director():
+    '''Extracts all movies from csv and stores them in a dictionary
+    where keys are directors, and values is a list of movies (named tuples)'''
+    directors = defaultdict(list)
+    with open(MOVIE_DATA, newline='') as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            if row['title_year'] and int(row['title_year']) >= MIN_YEAR:
+                directors[row['director_name']].append(
+                    Movie(row['movie_title'], row['title_year'], float(row['imdb_score'])))
+    return directors
+
+
+def get_average_scores(directors):
+    '''Filter directors with < MIN_MOVIES and calculate averge score'''
+    new_dict = {}
+    for k, v in directors.copy().items():
+        if len(v) >= MIN_MOVIES:
+            new_dict[(k, _calc_mean(directors[k]))] = v
```

Comment:
this is something you can do, but maybe should not do, its hard to maintain a dictionary with compound keys like this. You may come up with a different solution where you change the value to something like (score, movies), that is a Tuple with a float and a list